presto-tests/src/test/java/com/facebook/presto/tests/TestIsNotDistinctFrom.java

```diff
@@ -0,0 +1,119 @@
+            MaterializedResult result = queryRunner.execute("select * from (values nan(), nan(), nan()) group by 1");
+            assertTrue(Double.isNaN((Double) result.getOnlyValue()));
+        }
+    }
+
+    @Test
+    public void testRow()
+    {
+        try (QueryRunner queryRunner = createQueryRunner()) {
+            String query =
+                    "SELECT a.col1[1].col0, SUM(a.col0), SUM(a.col1[1].col1), SUM(a.col1[2].col0), SUM(a.col2.col1) " +
+                            "FROM (VALUES ROW(CAST(ROW(2.2, ARRAY[row(31, 4.2E0), row(22, 4.2E0)], row(5, 4.0E0)) " +
+                            "AS ROW(col0 double, col1 array(row(col0 integer, col1 double)), col2 row(col0 integer, col1 double)))), " +
+                            "ROW(CAST(ROW(1.0, ARRAY[row(31, 4.5E0), row(12, 4.2E0)], row(3, 4.1E0)) AS " +
+                            "ROW(col0 double, col1 array(row(col0 integer, col1 double)), col2 row(col0 integer, col1 double)))), " +
+                            "ROW(CAST(ROW(3.1, ARRAY[row(41, 3.1E0), row(32, 4.2E0)], row(6, 6.0E0)) AS " +
+                            "ROW(col0 double, col1 array(row(col0 integer, col1 double)), col2 row(col0 integer, col1 double)))), " +
+                            "ROW(CAST(ROW(3.3, ARRAY[row(41, 3.1E0), row(32, 4.2E0)], row(6, 6.0E0)) AS " +
+                            "ROW(col0 double, col1 array(row(col0 integer, col1 double)), col2 row(col0 integer, col1 double)))) ) t(a) " +
+                            "GROUP BY a.col1[1]";
+            MaterializedResult result = queryRunner.execute(query);
+            assertEquals(result.getRowCount(), 3);
+        }
+    }
+
+    @Test
+    public void testArray()
+    {
+        try (QueryRunner queryRunner = createQueryRunner()) {
+            String query = "select a from (values (array[nan(),2,3]), (array[nan(), 2, 3])) t(a) group by 1";
+            MaterializedResult result = queryRunner.execute(query);
+            assertEquals(result.getRowCount(), 1);
+        }
+    }
+
+    @Test
+    public void testExcept()
+    {
+        try (QueryRunner queryRunner = createQueryRunner()) {
+            String query = "SELECT NULL, NULL EXCEPT SELECT NULL, NULL FROM (values (1, 2)) t(a, b)";
+            MaterializedResult result = queryRunner.execute(query);
+            assertEquals(result.getRowCount(), 0);
+        }
+    }
+    @Test
+    public void testMap()
+    {
+        try (QueryRunner queryRunner = createQueryRunner()) {
+            String query = "SELECT MAP_KEYS(x)[1] FROM (VALUES MAP(ARRAY['a'], ARRAY[ARRAY[1]]), MAP(ARRAY['b'], ARRAY[ARRAY[2]])) t(x) GROUP BY x";
+            MaterializedResult result = queryRunner.execute(query);
+            assertEquals(result.getRowCount(), 2);
+        }
+    }
+
+    private static QueryRunner createQueryRunner()
+    {
+        return new LocalQueryRunner(testSessionBuilder()
+                .setSystemProperty(SystemSessionProperties.PARSE_DECIMAL_LITERALS_AS_DOUBLE, "true")
```

Comment:
Don't do this. Write your double literals as `1e0` to force it into a binary floating point literal.